open-policy-agent: 0.70.0 -> 1.1.0, matrix-authentication-service: 0.12.0 -> 0.13.0

charlieegan3 opened the pull request:
Hope this is the correct process, we released the new version of OPA at the end of last year and the bot doesn't seem to have picked it up - perhaps because it's a major version?

The release notes are here: https://github.com/open-policy-agent/opa/releases/tag/v1.0.0, this release's main changes are:

- makes the Rego v1 syntax the default. Users of v0 rego now need to use backwards compatibility mode.
- deprecated builtins have been removed.
- The OPA server now listens on loopback rather than all interfaces.
- The packages have been moved under `v1` to allow v0 SDK use to not be distrupted. This requires a minor change in the package.nix file here to remove the same test in a new location.

Upgrade process is outlined here: https://www.openpolicyagent.org/docs/latest/v0-upgrade/

V0 compatible functionality outlined here: https://www.openpolicyagent.org/docs/latest/v0-compatibility/

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux [here](https://github.com/NixOS/nixpkgs/pull/374272#issuecomment-2639785936)
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Adda0 commented:
It seems that the bot did pick it up, but there were some issues with the update. Feel free to have a look at the [logs](https://nixpkgs-update-logs.nix-community.org/open-policy-agent/).

charlieegan3 commented:
Oh interesting, it looks like the `matrix-authentication-service` package is failing. I'll see what I can do about getting that updated.

charlieegan3 commented:
Hey @teutat3s! I opened this PR to get the OPA package updated, but it seems there's an issue with the `matrix-authentication-service` package too. (https://github.com/NixOS/nixpkgs/pull/369555#issue-2763876650). Might you have some more familiarity with that project to dig into what might be wrong? I don't have a `x86_64-linux` laptop to replicate this easily. Please let me know if I can help with any of the OPA details 🙂!

06kellyjac commented:
Hi @charlieegan3
Sorry for the delay. I'll take a quick look at the `matrix-authentication-service` issue since I have a fair few `x86_64-linux` boxes.

If you could make a couple of changes that'd be great:

* Please add a note about the breaking changes going from v0 to v1 in the 25.06 release notes. Doesn't have to be every detail but linking out to resources is helpful.
* Maybe bump to the new 1.0.1

charlieegan3 committed:
open-policy-agent: 0.70.0 -> 1.1.0

Signed-off-by: Charlie Egan <charlieegan3@users.noreply.github.com>

charlieegan3 commented:
Thanks @06kellyjac! appreciate it. My guess is that some updates to that package will be required, happy to help there but just not had time to work out what's going on in the project yet. Happy to spend some time next week on it if needs be.

I've updated this PR to be 1.0.1 and added release note too. I left the note as v1.0.0+ since there might be some follow on minor releases before the nixpkgs release goes out.

charlieegan3 commented:
Unsure what to do about https://github.com/NixOS/nixpkgs/pull/369555 in the meantime.

06kellyjac commented:
We can ignore the bot PR for now or draft it. :slightly_smiling_face: 

The build error is due to the v0 v1 migration.

We can either try pull the commits which move to `v1` and see if that works

https://github.com/element-hq/matrix-authentication-service/commit/881c6df5cc1719ac86efe42c3f49bf898f3fd243

Or I can make a separte patch just based on the `opa fmt --v0-v1 -w ./policies` output. Or maybe attaching `--v0-compatible` to the build would work

charlieegan3 commented:
Cool, good that they've already done the upgrade to v1. That at least makes things a bit easier.

> We can either try pull the commits which move to v1 and see if that works

If this is an option, it seems like a good one. Or we could encourage them to cut a release with those Rego changes too?

06kellyjac commented:
Here's a standalone patch if you want: https://gist.github.com/06kellyjac/7038b3d0826f1a1dcae57044a4521260 . Up to you :slightly_smiling_face: 

Id say asking for a fresh release might be worth it but we could be waiting quite a long time & there could be other changes to account for in the package

charlieegan3 commented:
Ok, I think I've done the right thing in 8c6dee060240791d24d7952176cf36ee145a90cf 🙏 I've got to head but will check back in Monday if this is still needing attention. Thanks for the help here!

teutat3s commented:
Thank you for starting to work on this. :raised_hands:  There are `v0.13.0` [release candidates](https://github.com/element-hq/matrix-authentication-service/releases/tag/v0.13.0-rc.1) for a new version of MAS with the required commit. If this can wait a bit, I'd like to get both MAS and OPA versions bumped together. If not, also fine by me.

charlieegan3 commented:
Hey @teutat3s! Thanks for sharing, I'd not seen that. Do you know yet when that release might be published? I agree it'd be preferrable to get both packages updated 'correctly' at the same time.

charlieegan3 commented:
Hey @teutat3s, I see there's a new release candidate now. Do you know if the release is going to be going out today or tomorrow? If we think it might be later, perhaps we can proceed with the patch here and update it again when the release is out?

teutat3s commented:
I don't know the answer to that question. I am not blocking this PR.

06kellyjac reviewed (approved):
LGTM

teutat3s commented:
We have a MAS release : )

SuperSandro2000 commented on nixos/doc/manual/release-notes/rl-2505.section.md:
```diff
@@ -260,6 +260,14 @@
   To keep the old behavior for a site `example.com`, set `services.caddy.virtualHosts."example.com".hostName = "http://example.com"`.
   If you set custom Caddy options for a DokuWiki site, migrate these options by removing `http://` from `services.caddy.virtualHosts."http://example.com"`.
 
+- `open-policy-agent` has has been updated to v1.0.0+. This major release makes
+  the `rego.v1` syntax the default. This is a breaking change for those using v0
+  Rego. See the
+  [upgrade documentation](https://www.openpolicyagent.org/docs/v1.0.1/v0-upgrade/)
+  for more details. For those unable to upgrade yet, there is a
+  [v0 compatibility mode](https://www.openpolicyagent.org/docs/v1.0.1/v0-compatibility/)
+  available too.
```
```suggestion
- `open-policy-agent` has has been updated to 1.0.0+.
  This major release makes the `rego.v1` syntax the default.
  This is a breaking change for those using v0 Rego.
  See the [upgrade documentation](https://www.openpolicyagent.org/docs/v1.0.1/v0-upgrade/) for more details.
  For those unable to upgrade yet, there is a [v0 compatibility mode](https://www.openpolicyagent.org/docs/v1.0.1/v0-compatibility/)
  available too.
```

charlieegan3 committed:
matrix-authentication-service: 0.12.0 -> 0.13.0

This is included here as, since OPA is a dependency of MAS, the versions
of these two packages need to be updated together. (since 0.12.0 MAS
does not work with 1.0.0 OPA).

I have removed the sea-query check as it's no longer loading a special
case rc tag from a git repo.

Signed-off-by: Charlie Egan <charlieegan3@users.noreply.github.com>

SuperSandro2000 commented:
Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions) when naming your commits.

charlieegan3 commented:
Thanks for the prompt, have squashed and update now. Hope that looks ok.

charlieegan3 commented:
I think we can remove the wait-for-upstream label now too?

teutat3s reviewed (commented):
For MAS, we'll need to update the npmDeps hash and cargo hash as well.

teutat3s commented:
Thank you for tackling the MaS update as well. I hope we don't need the Cargo.lock anymore. Please try using `cargoHash` and removing the `Cargo.lock` file.

charlieegan3 commented:
Happy to 🙂 - hope that looks better now?

charlieegan3 commented:
I've added that, but seem to have a new failure now 🤔 

https://github.com/NixOS/nixpkgs/actions/runs/13175826112/job/36775098955?pr=374272

```
Run # Get the latest eval.yml workflow run for the PR's target commit
Comparing against "https://github.com/NixOS/nixpkgs/actions/runs/13175631325"
Workflow was not successful (conclusion: failure), cannot make comparison
Error: Process completed with exit code 1.
```

charlieegan3 commented:
The checks seem to be ok after the rebase, I can also run the build locally now to completion:

```
nix build --no-link --print-out-paths -L .#matrix-authentication-service
```

SuperSandro2000 commented on pkgs/by-name/ma/matrix-authentication-service/package.nix:
```diff
@@ -17,29 +17,25 @@
-      "sea-query-0.32.0-rc.1" = "sha256-Q/NFiIBu8L5rQj4jwcIo8ACmAhLBy4HSTcJv06UdK8E=";
-    };
-  };
+  cargoHash = "sha256-iUAdVnIeBxURqEmqom8MTMLuNnggDxoeVHVlNN8dyJ0=";
```
```suggestion
  cargoHash = "sha256-0RqCgIiH7O4bAuzy1To0+8wfvZWnZ8irZNSCthaKqAk=";
```

SuperSandro2000 committed:
matrix-authentication-service: fix vendor hash

teutat3s commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 374272`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>matrix-authentication-service</li>
    <li>open-policy-agent</li>
  </ul>
</details>

charlieegan3 commented:
Thanks everyone for all the back and forth here and for keeping me right! Much appreciated.
